python/tests/operations/unique_metadata.py

```diff
@@ -0,0 +1,588 @@
+import os
+import unittest
+
+import numpy as np
+
+import equistore.io
+import equistore.operations as fn
+from equistore import Labels, TensorBlock, TensorMap
+
+
+DATA_ROOT = os.path.join(os.path.dirname(__file__), "..", "data")
+TEST_FILE = "qm7-spherical-expansion.npz"
+
+
+class TestUniqueMetadata(unittest.TestCase):
+    """Finding unique_metadata metadata along a given axis of a TensorMap or
+    TensorBlock"""
+
+    def setUp(self):
+        self.tensor1 = test_tensor_map()  # Test case 1
+        self.tensor2 = test_large_tensor_map()  # Test case 2
+        self.tensor3 = equistore.io.load(  # Test case 3
+            os.path.join(DATA_ROOT, TEST_FILE),
+            use_numpy=True,
+        )
+
+    def test_unique_indices_block(self):
+        # Test case 1
+        samples = [0, 1, 2, 5]
+        properties = [0]
+        target_samples = Labels(
+            names=["samples"], values=np.array([[s] for s in samples])
+        )
+        target_properties = Labels(
+            names=["properties"], values=np.array([[p] for p in properties])
+        )
+        actual_samples = fn.unique_metadata_block(
+            self.tensor1.block(3), "samples", "samples"
+        )
+        actual_properties = fn.unique_metadata_block(
+            self.tensor1.block(3), "properties", "properties"
+        )
+        self.assertTrue(
+            fn._utils._labels_equal(target_samples, actual_samples, exact_order=True)
+        )
+        self.assertTrue(
+            fn._utils._labels_equal(
+                target_properties, actual_properties, exact_order=True
+            )
+        )
```

Comment:
Can you maybe split this into individual test functions. Preferably each unittests only has one assert which can fail. I know that this sometimes hard but this helps debugging a lot.